test/metabase/query_processor/pivot_test.clj

```diff
@@ -800,3 +800,33 @@
                 [nil 1 3]]
                (mt/rows
                 (qp.pivot/run-pivot-query query))))))))
+
+#_{:clj-kondo/ignore [:metabase/disallow-hardcoded-driver-names-in-tests]}
+(deftest pivot-table-from-native-mongo-query-test
+  (testing "#64124"
+    (mt/test-drivers #{:mongo}
+      (mt/dataset test-data
+        (let [query {:database (mt/id)
+                     :type     :query
+                     :query    {:source-table "card__1"
+                                :aggregation  [[:count]]
+                                :breakout     [[:field "PRODUCT_ID" {:base-type :type/Integer}]
+                                               [:field "QUANTITY" {:base-type :type/Integer}]]}}
+              pivot-settings {:pivot_table.column_split {:rows    ["PRODUCT_ID"]
+                                                         :columns ["QUANTITY"]
+                                                         :values  ["count"]}}]
+          (mt/with-metadata-provider
+            (lib.tu/mock-metadata-provider
+             (mt/metadata-provider)
+             {:cards [{:id 1
+                       :dataset-query {:type     :native
+                                       :native   {:collection "orders"
+                                                  :query      "[{\"$project\": {\"_id\": 0, \"PRODUCT_ID\": 1, \"QUANTITY\": 1}}]"}
+                                       :database (mt/id)}
+                       :database-id (mt/id)
+                       :result-metadata [{:name "PRODUCT_ID" :base_type :type/Integer}
+                                         {:name "QUANTITY" :base_type :type/Integer}]}]})
```

Comment:
I'm like 99% sure this is not doing anything here, it's not even valid legacy result metadata since `display_name` has always been a required key and we don't populate it automatically. You should be able to remove it